kuma/wiki/views/document.py

```diff
@@ -1107,3 +1107,43 @@ def _document_api_PUT(request, document_slug, document_locale):
         resp.status_code = 205
 
     return resp
+
+
+@never_cache
+@require_GET
+def doc_api(request, locale, path):
+    """
+    Return a JSON object that includes document content and metadata
+    for the document specified by the locale and path. Raises a 404
+    error if no such document exists. This is an API with URL
+    /api/v1/doc/<locale>/<path>
+    """
+    document = get_object_or_404(Document, locale=locale, slug=path)
+
+    return JsonResponse({
+        'locale': locale,
+        'path': path,
+        'id': document.id,
+        'title': document.title,
+        'summary': document.get_summary_html(),
+        'language': document.language,
+        'absoluteURL': document.get_absolute_url(),
+        'redirectURL': document.get_redirect_url(),
+        'editURL': document.get_edit_url(),
+        'bodyHTML': document.get_body_html(),
+        'quickLinksHTML': document.get_quick_links_html(),
+        'tocHTML': document.get_toc_html(),
+        'parents': [
+            {
+                'url': d.get_absolute_url(),
+                'title': d.title
+            } for d in document.parents
+        ],
+        'translations': [
+            {
+                'locale': t.locale,
+                'url': t.get_absolute_url(),
+                'title': t.title
+            } for t in document.other_translations
```

Comment:
Since you're only using the `locale`, `url`, and `title` fields of each of the translations, it's more efficient to replace this:
```
document.other_translations
``` 
with this:
```
document.get_other_translations(fields=('locale', 'url', 'title'))
```
in order to avoid the cost of transferring all of the other fields (especially, the content fields like `html` and `rendered_html`) from the database (the `other_translations` property is only cached on the document instance, so doesn't help us in this context).

Reply:
Done